fakeredis/_server.py

```diff
@@ -1426,16 +1427,22 @@ def set_(self, key, value, *args):
                 if px <= 0:
                     raise SimpleError(INVALID_EXPIRE_MSG.format('set'))
                 i += 2
+            elif casematch(args[i], b'keepttl'):
+                keepttl = True
+                i += 1
             else:
                 raise SimpleError(SYNTAX_ERROR_MSG)
-        if (xx and nx) or (px is not None and ex is not None):
+        if (xx and nx) or ((px is not None) + (ex is not None) + keepttl > 1):
```

Comment:
[`SET`](https://redis.io/commands/set) command documentation suggests that `KEEPTTL` option can be set next to `PX` or `EX` options, but in reality redis denies such requests. This behavior seems reasonable so I believe this is an error in the documentation and thus I enforce the actual (not the documented) behavior on the case of conflicting expiration options.